plugins/idea/src/main/java/com/oracle/plugin/jtreg/util/JTRegUtils.java

```diff
@@ -315,6 +320,52 @@ public static boolean isTestNGImport(PsiImportStatement importStatement) {
         return qualifiedName != null && qualifiedName.startsWith("org.testng");
     }
 
+    /**
+     * Judge whether the given file is a junit test.
+     */
+    public static boolean isJUnitTestData(Project project, VirtualFile file) {
+        return isJUnitTestData(PsiUtil.getPsiFile(project, file));
+    }
+
+    /**
+     * Judge whether the given file has the junit import statement.
+     */
+    public static boolean isJUnitTestData(PsiFile file) {
+        if (file instanceof PsiJavaFile) {
+            return Stream.of(((PsiJavaFile) file).getImportList().getImportStatements())
+                    .anyMatch(JTRegUtils::isJunitImport);
+        }
+        return false;
+    }
+
+    /**
+     * Judge whether the statement is a junit import statement.
+     */
+    public static boolean isJunitImport(PsiImportStatement importStatement) {
+        String qualifiedName = importStatement.getQualifiedName();
+        return qualifiedName != null && qualifiedName.startsWith("org.junit");
+    }
+
+    /**
+     * Create (if not existing) and get the jtreg project library.
+     */
+    public static Library createJTRegLibrary(Project project, String oldDir, String newDir) {
```

Comment:
Wouldn't it be better to split this into a function that creates the library, and one that updates it? I find it odd that the file listener is calling this with oldDir == newDir. Maybe just add option for oldDir to be null, so that we only drop the old root if one is given?

Reply:
I adjusted the api, added `update` method and delegated the `create` method to it. And I also fixed the possible NPE and avoided the duplicated lib entries.